### PR TITLE
chore: use RxJS for wait for Navigation

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: packages/puppeteer-core/src/bidi/Frame.ts [hunk 1]
```diff
@@ -201,48 +210,32 @@ export class BidiFrame extends Frame {
...
           this.#context,
-          waitUntilEvent,
-          () => {
-            return true;
-          },
-          timeout,
-          this.#abortDeferred.valueOrThrow()
+          Bidi.ChromiumBidi.BrowsingContext.EventNames.NavigationStarted
+        ).pipe(
+          switchMap(() => {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/Frame.ts:226]: how does it work? would it wait for NavigationStarted and the waitUntilEvent event after that? or would it be more like the promise.all in the previous version

Comment (OrKoN) [reply packages/puppeteer-core/src/bidi/Frame.ts:226]: the spec guarantees that that the waitUntil will happen after the nav started, right?

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/bidi/Frame.ts:226]: Yeah switch map creates a new observable each time the previous fires an invalidates the previous.
I will have to look into the spec but I believe (Both events are defined on the HTML spec so should be clear what the order is).

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/bidi/Frame.ts:226]: Looked at the HTML spec the navigation started event is emitted very early on the navigation lifecycle before even the `HTML's` response has completed.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**